impl/src/main/java/org/ehcache/internal/store/heap/OnHeapStore.java

```diff
@@ -1376,18 +1401,23 @@ private void expireMapping(final K key, final ValueHolder<V> value) {
 
     long expirationTime = duration.isForever() ? ValueHolder.NO_EXPIRE : safeExpireTime(now, duration);
 
-    eventSink.created(key, value);
-
-    return makeValue(key, value, now, expirationTime, this.valueCopier);
+    OnHeapValueHolder<V> holder = null;
+    try {
+      holder = makeValue(key, value, now, expirationTime, this.valueCopier);
+      eventSink.created(key, value);
+    } catch (LimitExceededException e) {
+      LOG.warn(e.getMessage());
+    }
+    return holder;
   }
 
-  private OnHeapValueHolder<V> importValueFromLowerTier(K key, ValueHolder<V> valueHolder, long now) {
+  private OnHeapValueHolder<V> importValueFromLowerTier(K key, ValueHolder<V> valueHolder, long now) throws LimitExceededException {
     V realValue = valueHolder.value();
     Duration expiration = expiry.getExpiryForAccess(key, realValue);
-    return cloneValueHolder(key, valueHolder, now, expiration);
+    return cloneValueHolder(key, valueHolder, now, expiration, false);
   }
 
-  private OnHeapValueHolder<V> cloneValueHolder(K key, ValueHolder<V> valueHolder, long now, Duration expiration) {
+  private OnHeapValueHolder<V> cloneValueHolder(K key, ValueHolder<V> valueHolder, long now, Duration expiration, boolean dontSize) throws LimitExceededException {
```

Comment:
Can you invert the added boolean parameter - so that `true` means sizing happens, `false` means it does not.